src/custom/hooks/useWrapCallback.ts

```diff
@@ -37,222 +28,191 @@ export enum WrapType {
-      return <Trans>Enter {wrapped?.symbol} amount</Trans>
-
-    case WrapInputError.INSUFFICIENT_NATIVE_BALANCE:
-      return <Trans>Insufficient {native?.symbol} balance</Trans>
-    case WrapInputError.INSUFFICIENT_WRAPPED_BALANCE:
-      return <Trans>Insufficient {wrapped?.symbol} balance</Trans>
-  }
-} */
-
 /**
  * Given the selected input and output currency, return a wrap callback
- * @param inputCurrency the selected input currency
- * @param outputCurrency the selected output currency
- * @param typedValue the user input value
  */
-export default function useWrapCallback(
-  openTransactionConfirmationModal: (message: string, operationType: OperationType) => void,
-  closeModals: () => void,
-  inputCurrency?: Currency | null,
-  outputCurrency?: Currency | null,
-  inputAmount?: CurrencyAmount<Currency>,
-  isEthTradeOverride?: boolean
-): WrapUnwrapCallback {
-  const { chainId: connectedChainId, account } = useWalletInfo()
-  const chainId = supportedChainId(connectedChainId)
-  const wethContract = useWETHContract()
-  const balance = useCurrencyBalance(account ?? undefined, inputCurrency)
-  const wrappedBalance = useCurrencyBalance(account ?? undefined, inputCurrency?.wrapped)
-  // we can always parse the amount typed as the input currency, since wrapping is 1:1
-  /* const inputAmount = useMemo(
-    () => tryParseCurrencyAmount(typedValue, inputCurrency ?? undefined),
-    [inputCurrency, typedValue]
-  ) */
-  const addTransaction = useTransactionAdder()
+export function useWrapCallback(
+  inputAmount: CurrencyAmount<Currency> | undefined,
+  openTransactionConfirmationModal: (message: string, operationType: OperationType) => void
+): WrapUnwrapCallback | null {
+  const context = useWrapUnwrapContext(inputAmount, openTransactionConfirmationModal)
+
+  if (!context) {
+    return null
+  }
 
-  return useMemo(() => {
-    if (!wethContract || !chainId || !inputCurrency || !outputCurrency) return NOT_APPLICABLE
-    const weth = WRAPPED_NATIVE_CURRENCY[chainId]
-    if (!weth) return NOT_APPLICABLE
-
-    const isWrappingEther = inputCurrency.isNative && (isEthTradeOverride || weth.equals(outputCurrency))
-    const isUnwrappingWeth = weth.equals(inputCurrency) && outputCurrency.isNative
-    // is an native currency trade but wrapped token has enough balance
-    const hasEnoughWrappedBalanceForSwap =
-      isEthTradeOverride && wrappedBalance && inputAmount && !wrappedBalance.lessThan(inputAmount)
-
-    if ((!isWrappingEther && !isUnwrappingWeth) || hasEnoughWrappedBalanceForSwap) {
-      return NOT_APPLICABLE
-    } else {
-      return _getWrapUnwrapCallback({
-        chainId,
-        isWrap: isWrappingEther,
-        balance,
-        inputAmount,
-        addTransaction,
-        wethContract,
-        openTransactionConfirmationModal,
-        closeModals,
-      })
-    }
-  }, [
+  return (params?: WrapUnwrapCallbackParams) => {
+    return wrapUnwrapCallback(context, params || { useModals: true })
```

Comment:
why not give directly `wrapUnwrapCallback` default `{ useModals: true }` as the `params`?

Reply:
Fixed